rest/config.go

```diff
@@ -664,33 +658,32 @@ func decodeAndSanitiseConfig(r io.Reader, config interface{}) (err error) {
 	return base.WrapJSONUnknownFieldErr(err)
 }
 
-func (config *ServerConfig) setupAndValidateDatabases() []error {
+func (config *ServerConfig) setupAndValidateDatabases() (errs *multierror.Error) {
 	if config == nil {
 		return nil
 	}
 
 	for name, dbConfig := range config.Databases {
 
 		if err := dbConfig.setup(name); err != nil {
-			return []error{err}
+			return multierror.Append(errs, err)
```

Comment:
One of the advantages of multierror is that it implements error.  Given that, we could take advantage of this and just have setupAndValidateDatabases return error, and just 'return err' here.  What do you think?

Reply:
Yeah, that would be more cleaner - setupAndValidateDatabases has been modified to return an error and cast it back to multierror.Error in places where the caller knows about the error and needs to validate the error. This is applicable in all other places where we are returning multierror.Error.

Reply:
The usage of setupAndValidateDatabases in serverMain would work if you just did `return err` here, instead of wrapping a single error in multierror.  The only other usages are unit tests - are any of those broken by just doing `return err` here?  If so, is it possible to tweak the unit tests that are specifically checking for dbConfig.setup errors so that this can be simplified?

Reply:
Yeah, the reason for returning multi-error here was that to avoid the potential failure in TestSetupAndValidateDatabases while casting the error back to multi-error. But I see what you suggest here and it is reasonable to tweak the TestSetupAndValidateDatabases test.